core/controllers/learner_playlist.py

```diff
@@ -27,12 +27,31 @@ class LearnerPlaylistHandler(base.BaseHandler):
     @acl_decorators.can_access_learner_dashboard
     def post(self, activity_type, activity_id):
         position_to_be_inserted_in = self.payload.get('index')
+
+        playlist_limit_exceeded = False
+        belongs_to_completed_or_incomplete_list = False
```

Comment:
That said, on reading further, I'm not sure this is the right modelling. It looks like you should be returning some kind of error object/constant according to the error you actually want to show in the dashboard. This would also be cleaner than returning tuples all over the place.